portfolio/src/main/webapp/script.js

```diff
@@ -38,4 +38,14 @@ function showProjects(){
     projectContainer.style.display = "none";
   }
 
+}
+
+async function sayHello(){
+    const response = await fetch("/hello");
+    const text = await response.text();
+
+    const greeting_container = document.getElementById('greeting-container');
+    new_text = text.replace("<h1>", "")
```

Comment:
JavaScript is weird so you _can_ use a variable without defining it, but it's usually a better idea to define it first. In other words, I would change this line to this:

```
let newText = text.replace('<h1>', '');
```

(I'd also consider using lowerCamelCase for the variable name and single quotes for strings, as per the style guide: https://google.github.io/styleguide/jsguide.html